frontends/web/src/components/backbutton/backbutton.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentPropsWithoutRef, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useEsc } from '@/hooks/keyboard';
+import { Button } from '@/components/forms/button';
+
+type TBackButton = Omit<ComponentPropsWithoutRef<'button'>, 'onClick'> & {
+  canGoBack?: boolean;
+};
+
+export const BackButton = ({
+  canGoBack,
+  children,
+  ...props
+}: TBackButton) => {
+  const navigate = useNavigate();
+  const handleBack = useCallback(() => {
+    if (canGoBack) {
+      navigate(-1);
+    }
+  }, [canGoBack, navigate]);
+
+  useEsc(handleBack);
+
+  return (
+    <Button
+      secondary
+      onClick={handleBack}
+      disabled={!canGoBack}
+      {...props}>
```

Comment:
Looks like `disabled` can be passed in these props too or not? Could be confusing having both canGoBack and disabled. Imho I'd just remove `canGoBack` and use `disabled` instead where needed.

Reply:
makes sense